up: 0.3.1 -> 0.3.2

Ma27 opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

New release with several new keyboard shortcuts to navigate between
piped commands: https://github.com/akavel/up/releases/tag/v0.3.2

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Ma27 committed:
up: 0.3.1 -> 0.3.2

New release with several new keyboard shortcuts to navigate between
piped commands: https://github.com/akavel/up/releases/tag/v0.3.2

marsam reviewed (commented):
looks like up uses go modules, would you mind updating it to [`buildGoModule`](https://nixos.org/nixpkgs/manual/#ssec-go-modules)?

tomberek reviewed (approved):
Just a data point: builds and runs on Ubuntu. No reverse dependencies.

Ma27 committed:
up: repackage with buildGoModule

Go modules[1] provide a set of sources and corresponding checksums.
These sources can be resolved automatically with Nix, so no go2nix is
required anymore.

[1] https://blog.golang.org/using-go-modules

Ma27 commented:
@marsam done :)
